src/python/pants/releases/packages.py

```diff
@@ -241,22 +242,43 @@ def check_ownership(i, package):
     sys.exit(1)
 
 
-if sys.argv[1:] == ["list"]:
-  print('\n'.join(package.name for package in sorted(all_packages())))
-elif sys.argv[1:] == ["list", "--with-packages"]:
-  print('\n'.join('{} {} {}'.format(package.name, package.target, " ".join(package.bdist_wheel_flags)) for package in sorted(all_packages())))
-elif sys.argv[1:] == ["list-owners"]:
+def _create_parser():
+  parser = argparse.ArgumentParser()
+  subparsers = parser.add_subparsers(dest="command")
+  # list
+  parser_list = subparsers.add_parser('list')
+  parser_list.add_argument("--with-packages", action="store_true")
+  # list-owners
+  subparsers.add_parser("list-owners")
+  # check-ownership
+  subparsers.add_parser("check-ownership")
+  # build_and_print
+  parser_build_and_print = subparsers.add_parser("build_and_print")
+  parser_build_and_print.add_argument("version")
+  return parser
+
+
+args = _create_parser().parse_args()
+
+if args.command == "list":
```

Comment:
Rather than this huge `if else` block, I considered modifying this to be a dictionary lookup against `args.command` like this:

```python
func = {"list": run_list,
  "list-owners": run_list_owners,
  ...
}[args.command]
func()
```

Didn't try it out because it would require turning all the below logic into `def` functions—Python doesn't have multiline lambadas—and didn't think you all want want the extra noise / LOC. I don't think we have to worry about a `KeyError` or invalid arg combination like we currently do with the `else` block, because `argparse` will handle that for us already. Let me know if you'd like me to make that change.

Reply:
Yeah, not an obvious win afaict. This is a clean / scoped change as is.